src/main/java/net/javadiscord/javabot/systems/help/AutoCodeFormatter.java

```diff
@@ -0,0 +1,102 @@
+package net.javadiscord.javabot.systems.help;
+
+import lombok.AllArgsConstructor;
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.javadiscord.javabot.data.config.BotConfig;
+import net.javadiscord.javabot.systems.moderation.AutoMod;
+import net.javadiscord.javabot.systems.user_preferences.UserPreferenceService;
+import net.javadiscord.javabot.systems.user_preferences.model.Preference;
+import net.javadiscord.javabot.util.ExceptionLogger;
+import net.javadiscord.javabot.util.WebhookUtil;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.annotation.Nonnull;
+import java.util.Objects;
+
+/**
+ * Handles auto-formatting of code in help-channels.
+ * A non-listener class since it has to be called strictly after {@link HelpListener} to not send its messages before channel reserving.
+ */
+@AllArgsConstructor
```

Comment:
`@RequiredArgsConstructor` makes more sense here since one might want to include other fields for internal state or similar.